### PR TITLE
Pbi 1216

### PR DESCRIPTION
**Pull request by aeng7**
-ima-vtc integration for non-shared stream elements (desktop webkit, Android)
-shared stream implementation will come in the near future

Notes/questions:
-what should default volume and currentTime be if required components are not setup yet (for getVolume and getCurrentTime)?

### CODE REVIEW

**Comment by angieisrani** [comment]
Good questions:  Assume volume=1.0 (100%) if not configured.  Assume currentTime=0 if not played yet unless setInitialTime called.

File: js/google_ima.js [hunk 1]
```diff
@@ -1414,13 +1518,241 @@ OO.Ads.manager(function(_, $)
...
+    this.maxSupportedElements = -1;
+
+    /**
+     * Returns the number of video elements currently instantiated.
+     * @public
+     * @method GoogleIMAVideoFactory#getCurrentNumberOfInstances
+     * @returns {int} The number of video elements created by this factory that have not been destroyed
+     */
+    this.getCurrentNumberOfInstances = function() {
+    };
```
Comment (angieisrani) [inline js/google_ima.js]: This function is being removed - you can take it out of here

**Comment by angieisrani** [comment]
So far looks good to me.  Mind walking through a couple pieces with me in person?

**Comment by aeng7** [comment]
Sure, will do.

**Comment by aeng7** [comment]
Support for multiple players has been implemented.

**Comment by aeng7** [comment]
Since I moved the old Ad Manager code inside a function, the diff is hard to parse. The main changes:
-changed default values for getCurrentTime and getVolume inside GoogleIMA
-added a playerId parameter to the GoogleIMA object's initialize method, which is used to store the instance inside registeredGoogleIMAManagers
-registeredGoogleIMAManagers will be used to retrieve instances when the Factory's create method is called. The create method now includes the playerId parameter

These changes were made to support multiple players on a page.

**Comment by aeng7** [comment]
You can also do a local diff and pull the OO.Ads.manager(); block of code outside of and above the outer anonymous function. The diff should be much more readable that way.

**Comment by angieisrani** [comment]
Looks great to me - merging

**Comment by angieisrani** [comment]
@aeng7 I missed one thing - can you update the template with the new parameter for the initialize function and open a new pull request?

**Comment by aeng7** [comment]
Will do.
